routes/index.js

```diff
@@ -2,12 +2,23 @@
  * Copyright (c) Jupyter Development Team.
  * Distributed under the terms of the Modified BSD License.
  */
+var debug = require('debug')('dashboard-proxy:server');
 var express = require('express');
+var multer = require('multer');
 var nbstore = require('../app/notebook-store');
-var debug = require('debug')('dashboard-proxy:server');
+var uploadNb = require('../app/upload-notebook');
 
 var router = express.Router();
 
+// multer setup
+var storage = multer.diskStorage({
+    destination: uploadNb.destination,
+    filename: uploadNb.filename
+});
+var upload = multer({
+    storage: storage
+});
+
```

Comment:
I would prefer that this code chunk (and require of `multer`, etc) be moved to ` app/upload-notebook.js`. Then in this file, we just require that file and set the `upload` object, to be used below. Makes the code cleaner and more modular.